### DISCUSSION TITLE
is it possible to disable icmpv6 request

### DISCUSSION THREAD

**Discussion by wrliao** [rtbrick/bngblaster#173] [Q&A] [answered]
Hi

I'm using below json on my dhcpv4 setup but found bngblaster is always sending icmpv6 which caused session can not be in eastablished state. is there any option to disable sending the icmpv6 ?

{
  "interfaces": {
    "access": [
      {
        "__comment__": "DHCP Client 1",
        "interface": "eth1",
        "type": "ipoe",
        "ipv6": "true",
        "vlan-mode": "1:1",
        "outer-vlan-min": 1001,
        "outer-vlan-max": 1001,
        "outer-vlan-step": 1,
        "inner-vlan-min": 1,
        "inner-vlan-max": 20,
        "inner-vlan-step": 1,
        "mac": "{random:mac}"
      },
      {
        "__comment__": "DHCP Client 2",
        "interface": "eth1",
        "type": "ipoe",
        "ipv6": "false",
        "vlan-mode": "1:1",
        "outer-vlan-min": 1002,
        "outer-vlan-max": 1002,
        "outer-vlan-step": 1,
        "inner-vlan-min": 1,
        "inner-vlan-max": 20,
        "inner-vlan-step": 1,
        "mac": "{random:mac}"
      },
      {
        "__comment__": "DHCP Client 3",
        "interface": "eth1",
        "type": "ipoe",
        "ipv6": "false",
        "vlan-mode": "1:1",
        "outer-vlan-min": 1003,
        "outer-vlan-max": 1003,
        "outer-vlan-step": 1,
        "inner-vlan-min": 1,
        "inner-vlan-max": 20,
        "inner-vlan-step": 1,
        "mac": "{random:mac}"
      }
    ]
  },
  "dhcp": {
    "enable": true,
    "broadcast": false 
  },
  "dhcpv6": {
    "enable": false, 
    "ia-na": false, 
    "ia-pd": false 
  },
  "access-line": {
    "agent-remote-id": "bng-sub.{session-global}",
    "agent-circuit-id": "0.0.0.0/0.0.0.0 eth 1:{session-global}"
  }
}

      ____   __   ____         _        __                                  ,/
     / __ \ / /_ / __ ) _____ (_)_____ / /__                              ,'/
    / /_/ // __// __  |/ ___// // ___// //_/                            ,' /
   / _, _// /_ / /_/ // /   / // /__ / ,<                             ,'  /_____,    
  /_/ |_| \__//_____//_/   /_/ \___//_/|_|                          .'____    ,'   
      ____   _   _  ______   ____   _               _                    /  ,'
     / __ ) / | / // ____/  / __ ) / /____ _ _____ / /_ ___   ____      / ,'
    / __  |/  |/ // / __   / __  |/ // __ `// ___// __// _ \ / ___/    /,'
   / /_/ // /|  // /_/ /  / /_/ // // /_/ /(__  )/ /_ /  __// /       / 
  /_____//_/ |_/ \____/  /_____//_/ \__,_//____/ \__/ \___//_/

Report:
==============================================================================
Test Duration: 139s
Sessions PPPoE: 0 IPoE: 60
Sessions established: 0/60
DHCPv6 sessions established: 0
Setup Time: 0 ms
Setup Rate: 0.00 CPS (MIN: 0.00 AVG: 0.00 MAX: 0.00)
Flapped: 0

Interface: eth1
------------------------------------------------------------------------------
  TX:                      2040 packets           217560 bytes
  TX Polled:                  0
  RX:                       183 packets            43938 bytes
  RX Protocol Error:          0 packets
  RX Unknown:                 0 packets
  RX Polled:             130142

Access Interface: eth1
  TX:                      2040 packets           217560 bytes
  RX:                       183 packets            43938 bytes
  RX Multicast:               0 packets                0 loss

Access Interface Protocol Packet Stats:
  ARP    TX:         60 RX:         60
  PADI   TX:          0 RX:          0
  PADO   TX:          0 RX:          0
  PADR   TX:          0 RX:          0
  PADS   TX:          0 RX:          0
  PADT   TX:          0 RX:          0
  LCP    TX:          0 RX:          0
  PAP    TX:          0 RX:          0
  CHAP   TX:          0 RX:          0
  IPCP   TX:          0 RX:          0
  IP6CP  TX:          0 RX:          0
  IGMP   TX:          0 RX:          0
  ICMP   TX:          0 RX:          0
  DHCP   TX:        300 RX:        123
  DHCPv6 TX:          0 RX:          0
  **ICMPv6 TX:       1680 RX:          0**
  IPv4 Fragmented       RX:          0

Access Interface Protocol Timeout Stats:
  LCP Echo Request:          0
  LCP Request:               0
  IPCP Request:              0
  IP6CP Request:             0
  PAP:                       0
  CHAP:                      0
  DHCP Request:            120
  DHCPv6 Request:            0
  **ICMPv6 RS:              1620**

**Comment by GIC-de** [comment]
But you still have IPv6 enabled for client 1, which per RFC triggers ICMPv6.
 
```
"comment": "DHCP Client 1",
"interface": "eth1",
"type": "ipoe",
"ipv6": "true", <<<<<<<
"vlan-mode": "1:1",
```

**Reply by wrliao** [reply]
I simplfied the json file with one outer range as below but still can see icmpv6.

{
  "interfaces": {
    "access": [
      {
        "__comment__": "DHCP Client 3",
        "interface": "eth1",
        "type": "ipoe",
        "ipv6": "false",
        "vlan-mode": "1:1",
        "outer-vlan-min": 1003,
        "outer-vlan-max": 1003,
        "outer-vlan-step": 1,
        "inner-vlan-min": 1,
        "inner-vlan-max": 20,
        "inner-vlan-step": 1,
        "mac": "{random:mac}"
      }
    ]
  },
  "dhcp": {
    "enable": true,
    "broadcast": false
  },
  "dhcpv6": {
    "enable": false,
    "ia-na": false,
    "ia-pd": false
  },
  "access-line": {
    "agent-remote-id": "bng-sub.{session-global}",
    "agent-circuit-id": "0.0.0.0/0.0.0.0 eth 1:{session-global}"
  }
}

Report:
==============================================================================
Test Duration: 6s
Sessions PPPoE: 0 IPoE: 10
Sessions established: 0/10
DHCPv6 sessions established: 0
Setup Time: 0 ms
Setup Rate: 0.00 CPS (MIN: 0.00 AVG: 0.00 MAX: 0.00)
Flapped: 0

Interface: eth1
------------------------------------------------------------------------------
  TX:                        80 packets            18060 bytes
  TX Polled:                  0
  RX:                        40 packets             7720 bytes
  RX Protocol Error:          0 packets
  RX Unknown:                 0 packets
  RX Polled:               5589

Access Interface: eth1
  TX:                        80 packets            18060 bytes
  RX:                        40 packets             7720 bytes
  RX Multicast:               0 packets                0 loss

Access Interface Protocol Packet Stats:
  ARP    TX:         10 RX:         20
  PADI   TX:          0 RX:          0
  PADO   TX:          0 RX:          0
  PADR   TX:          0 RX:          0
  PADS   TX:          0 RX:          0
  PADT   TX:          0 RX:          0
  LCP    TX:          0 RX:          0
  PAP    TX:          0 RX:          0
  CHAP   TX:          0 RX:          0
  IPCP   TX:          0 RX:          0
  IP6CP  TX:          0 RX:          0
  IGMP   TX:          0 RX:          0
  ICMP   TX:          0 RX:          0
  DHCP   TX:         50 RX:         20
  DHCPv6 TX:          0 RX:          0
  ICMPv6 TX:         20 RX:          0
  IPv4 Fragmented       RX:          0

Access Interface Protocol Timeout Stats:
  LCP Echo Request:          0
  LCP Request:               0
  IPCP Request:              0
  IP6CP Request:             0
  PAP:                       0
  CHAP:                      0
  DHCP Request:             20
  DHCPv6 Request:            0
  **ICMPv6 RS:                10**

**Reply by GIC-de** [reply]
Let me try to reproduce and maybe fix.

**Reply by GIC-de** [reply, marked as answer]
You set ` "ipv6": "false", ` instead of ` "ipv6": false, `.
